src/components/Dashboard/index.tsx

```diff
@@ -1,20 +1,75 @@
-import { Container, UserCard, UserList } from '../Dashboard/style';
-import data from '../../data.json';
+import { Container, UserCard, UserList, UserNavigation } from '../Dashboard/style';
+import { useQuery } from '@apollo/client';
+import { GET_USERS } from '../../utils/apollo-queries';
+import { useState } from 'react';
+
+interface UsersResponseData {
+  users: {
+    count: number;
+    nodes: {
+      id: number;
+      email: string;
+      name: string;
+    }[];
+    pageInfo: {
+      hasNextPage: boolean;
+      hasPreviousPage: boolean;
+      limit: number;
+      offset: number;
+    };
+  };
+}
 
 export const Dashboard: React.FC = () => {
+  const [offset, setOffset] = useState(0);
+  const limit = 12;
+  const { data, loading, error } = useQuery(GET_USERS, { variables: { offset, limit } });
+
+  const responseData: UsersResponseData = data;
+  const nodes = responseData?.users.nodes;
+  const pageInfo = responseData?.users.pageInfo;
+
   return (
     <Container>
       <h1>User List</h1>
-      <UserList>
-        {data.map((user) => {
-          return (
-            <UserCard>
-              <h2>{user.name}</h2>
-              <p>{user.email}</p>
-            </UserCard>
-          );
-        })}
-      </UserList>
+      {loading ? (
+        <img src='spinner.gif' height='100px' />
+      ) : error ? (
+        <h2>{error.message}</h2>
+      ) : (
+        <UserList>
+          {nodes?.map((user) => {
+            return (
+              <UserCard key={user.id}>
+                <h2>{user.name}</h2>
+                <p>{user.email}</p>
+              </UserCard>
+            );
+          })}
+        </UserList>
+      )}
+      {pageInfo?.hasPreviousPage ? (
+        <UserNavigation
+          onClick={() => {
+            setOffset(offset - limit);
+          }}
+        >
+          Prev Page
+        </UserNavigation>
```

Comment:
Aqui eu acho que quebrar a linha mais prejudicou do que ajudou:

```suggestion
        <UserNavigation onClick={() => setOffset(offset - limit)}>Prev Page</UserNavigation>
```